### PR TITLE
feat(web): server side filtering on audit log

### PR DESCRIPTION
**Pull request by balanceee**
Implement server-side filtering on audit-log.

Also:
- Refactor paginator

### CODE REVIEW

File: web/crux-ui/src/hooks/use-throttleing.ts [hunk 1]
```diff
@@ -19,11 +20,14 @@ export const useThrottling = (delay: number): ((action: VoidFunction) => void) =
       clearTimeout(schedule.current)
     }
 
-    schedule.current = setTimeout(() => action.trigger(), delay)
+    schedule.current = action.immediate
+      ? setTimeout(() => action.trigger(), 0)
```
Comment (polaroi8d) [inline web/crux-ui/src/hooks/use-throttleing.ts:24]: `setTimeout(() => action.trigger(), 0)` can be `setTimeout(() => action.trigger())` imho

File: web/crux-ui/src/models/audit.ts [hunk 2]
```diff
@@ -5,6 +5,14 @@ export type AuditLog = {
   info?: any
 }
 
+export type AuditLogListRequest = {
+  pageNumber: number
+  pageSize: number
+  keyword?: string
+  createdFrom?: string
+  createdTo: string
```
Comment (polaroi8d) [inline web/crux-ui/src/models/audit.ts:13]: If `createdFrom` optional, why `createdTo` is mandatory?

File: web/crux-ui/src/pages/audit-log.tsx [hunk 3]
```diff
@@ -3,82 +3,124 @@ import { BreadcrumbLink } from '@app/components/shared/breadcrumb'
...
-
   const onShowInfoClick = (logEntry: AuditLog) => setShowInfo(logEntry)
 
-  useEffect(() => {
-    pagination.setItems(filters.filtered)
-  }, [filters, pagination])
+  // Handlers
+  const onDateRangedChanged = dates => {
+    const [start, end] = dates
+    if (end !== null) end.setHours(23, 59, 59, 999)
```
Comment (polaroi8d) [inline web/crux-ui/src/pages/audit-log.tsx:109]: 🧙🏻

File: web/crux-ui/src/hooks/use-throttleing.ts (hunk 1 above)
Comment (balanceee) [reply web/crux-ui/src/hooks/use-throttleing.ts:24]: I will check it ty. Futhermore I want to discuss the whole solution with @m8vago later today

**Review by polaroi8d — APPROVED** [review]
LGTM (informal)

**Review by robot9706 — APPROVED** [review]
Discuss throttling with @m8vago, otherwise LGTM 😎

**Pull request merged**